lib/core/src/animation/animator/continuous.rs

```diff
@@ -0,0 +1,116 @@
+use crate::system::web::animation_frame_loop::AnimationFrameLoop;
+use crate::system::web::animation_frame_loop::AnimationFrameCallbackGuard;
+use super::AnimationCallback;
+
+use std::rc::Rc;
+use std::cell::RefCell;
+
+
+// ========================================
+// === ContinuousTimeAnimatorProperties ===
+// ========================================
+
+struct ContinuousTimeAnimatorProperties {
+    callback          : Box<dyn AnimationCallback>,
+    relative_start_ms : f32,
+    absolute_start_ms : Option<f32>,
+    callback_guard    : Option<AnimationFrameCallbackGuard>
+}
+
+
+
+// ==================================
+// === ContinuousTimeAnimatorData ===
+// ==================================
+
+struct ContinuousAnimatorData {
+    properties : RefCell<ContinuousTimeAnimatorProperties>
+}
+
+impl ContinuousAnimatorData {
+    fn new<F:AnimationCallback>(f:F) -> Rc<Self> {
+        let callback          = Box::new(f);
+        let relative_start_ms = 0.0;
+        let absolute_start_ms = None;
+        let callback_guard    = None;
+        let properties        = ContinuousTimeAnimatorProperties {
+            callback,
+            callback_guard,
+            relative_start_ms,
+            absolute_start_ms
+        };
+        let properties = RefCell::new(properties);
+        Rc::new(Self {properties})
+    }
+
+    fn on_animation_frame(&self, relative_time_ms:f32) {
+        (&mut self.properties.borrow_mut().callback)(relative_time_ms)
+    }
+}
+
+
+// === Setters ===
+
+impl ContinuousAnimatorData {
+    fn set_time(&self, time:f32) {
+        let mut properties = self.properties.borrow_mut();
+        properties.relative_start_ms = time;
+        properties.absolute_start_ms = None;
+    }
+
+    fn set_callback_guard(&self, callback_guard:Option<AnimationFrameCallbackGuard>) {
+        self.properties.borrow_mut().callback_guard = callback_guard;
+    }
+
+    fn set_absolute_start_ms(&self, absolute_start_ms:Option<f32>) {
+        self.properties.borrow_mut().absolute_start_ms = absolute_start_ms
+    }
+}
+
+
+// === Getters ===
+
+impl ContinuousAnimatorData {
+    fn relative_start_ms(&self) -> f32               { self.properties.borrow().relative_start_ms }
+    fn absolute_start_ms(&self) -> Option<f32>       { self.properties.borrow().absolute_start_ms }
```

Comment:
Strange alignment (big spacing befory body of decls). Let just make them multiline.